### PR TITLE
RUMM-649 fix clock skew

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Use the Datadog NTP servers to get a better timestamp on APM events

### Motivation

Our home made NTP mecanism still had some skews (around 500ms on average) which would give weird results when visualizing front to back flamegraphs.
We started using an NTP sync library, calling [Datadog's NTP servers](https://docs.datadoghq.com/integrations/ntp/) to get an accurate timestamp.

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/net/TracesOkHttpUploaderTest.kt [hunk 1]
```diff
@@ -43,6 +43,6 @@ internal class TracesOkHttpUploaderTest : DataOkHttpUploaderTest<TracesOkHttpUpl
     }
 
     override fun expectedPathRegex(): String {
-        return "\\/v1\\/input\\/$fakeToken$"
+        return "^\\/v1\\/input\\/$fakeToken$"
```
Comment (ncreated) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/net/TracesOkHttpUploaderTest.kt:46]: So, adding `batch_time` doesn't solve the problem for the APM Intake, and this is why we mix both approaches (`?batch_time` with NTP time)?

File: dd-sdk-android/apiSurface [hunk 2]
```diff
@@ -36,6 +36,10 @@ object com.datadog.android.DatadogEndpoint
   const val TRACES_EU: String
   const val RUM_US: String
   const val RUM_EU: String
+  const val NTP_0: String
+  const val NTP_1: String
+  const val NTP_2: String
+  const val NTP_3: String
```
Comment (ncreated) [inline dd-sdk-android/apiSurface:42]: Do we need to expose those in public interface? 🤔

File: dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/net/TracesOkHttpUploaderTest.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/net/TracesOkHttpUploaderTest.kt:46]: Yes

File: dd-sdk-android/apiSurface (hunk 2 above)
Comment (xgouchet) [reply dd-sdk-android/apiSurface:42]: Well they are part of the [official documentation](https://docs.datadoghq.com/integrations/ntp/), so it doesn't hurt to expose them.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/time/KronosTimeProviderTest.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        assertThat(result).isEqualTo(fakeDate.time)
+    }
+
+    @Test
+    fun `returns server time offset`() {
+        val now = System.currentTimeMillis()
+        val result = testedTimeProvider.getServerOffsetNanos()
+
+        val expectedOffset = TimeUnit.MILLISECONDS.toNanos(fakeDate.time - now)
+        assertThat(result).isCloseTo(expectedOffset, Offset.offset(10000000L))
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/time/KronosTimeProviderTest.kt:64]: `TimeUnit.MILLISECONDS.toNanos(1)` just for readability please.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt [hunk 4]
```diff
@@ -73,6 +77,19 @@ internal object CoreFeature {
...
 
+        kronosClock = AndroidClockFactory.createKronosClock(
+            appContext,
+            ntpHosts = listOf(
+                DatadogEndpoint.NTP_0,
+                DatadogEndpoint.NTP_1,
+                DatadogEndpoint.NTP_2,
+                DatadogEndpoint.NTP_3
+            ),
+            cacheExpirationMs = TimeUnit.MINUTES.toMillis(30),
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt:88]: No this means try to sync every 30 minutes, and never sync more than every 5 minutes. But in the end yes, on average this means the offset won't be refreshed during a 30 minutes interval. 
Usually people don't change their local time every 5 minutes, so we should be good, and I don't want the SDK to cause too many NTP requests for nothing.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/time/KronosTimeProviderTest.kt (hunk 3 above)
Comment (xgouchet) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/time/KronosTimeProviderTest.kt:64]: Sure

**Pull request merged**